source/as-promise.js

```diff
@@ -27,28 +28,34 @@ module.exports = options => {
 				return;
 			}
 
-			const {statusCode} = response;
 			const limitStatusCode = options.followRedirect ? 299 : 399;
 
 			response.body = data;
 
 			try {
 				for (const hook of options.hooks.afterResponse) {
 					// eslint-disable-next-line no-await-in-loop
-					response = await hook(response);
-
-					if (is.plainObject(response)) {
-						if (emitter.retry(response) === false) {
-							reject(new Error('Retry limit reached.'));
+					response = await hook(response, updatedOptions => {
+						if (emitter.retry(null) === false) {
+							throw new Error('Retry limit reached.');
 						}
-						return;
-					}
+
+						return asPromise(mergeOptions(options, {
+							...updatedOptions,
+							retry: {
+								retries: () => 0
+							},
```

Comment:
Why not just:

```js
retry: {
	retries: 0
},
```

Reply:
These options are normalized. `retry.retries` is a function.

Reply:
But doesn't `mergeOptions` normalize? I'm just asking, because if it doesn't normalize the `retry` there, it doesn't normalize `updatedOptions` either, which is problematic.

Reply:
yup, I forgot about it, sorry :p

Reply:
nah, you confused me! :P `mergeOptions` doesn't normalize. It just merges them.

Reply:
But the problem is still there. The users options needs to be normalized as if they called `got(options)`. That's what the user would expect.

Reply:
Will be :)